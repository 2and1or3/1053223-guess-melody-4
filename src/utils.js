const mockAudioNode = (element) => {
  if (element.type === `audio`) {
    return element;
  }

  return null;
};

export {mockAudioNode};
