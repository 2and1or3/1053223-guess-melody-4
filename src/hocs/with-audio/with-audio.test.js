import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import withAudio from './with-audio.js';
import {mockAudioNode} from '../../utils.js';

const MockComponent = (props) => {
  const {children} = props;

  return <div>{children}</div>;
};

MockComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

const MockComponentWrapped = withAudio(MockComponent);

it(`Render withAudio component`, () => {
  const tree = renderer
    .create(<MockComponentWrapped
      isPlaying = {false}
      onPlayButtonClick = {() => {}}
      src = {`path`}
    />, {
      createNodeMock: mockAudioNode
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
