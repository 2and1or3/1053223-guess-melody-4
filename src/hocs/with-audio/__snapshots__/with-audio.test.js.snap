// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withAudio component 1`] = `
<div>
  <audio />
</div>
`;
